proof_ipa.go

```diff
@@ -555,3 +555,62 @@ type bytesSlice []Stem
 func (x bytesSlice) Len() int           { return len(x) }
 func (x bytesSlice) Less(i, j int) bool { return bytes.Compare(x[i], x[j]) < 0 }
 func (x bytesSlice) Swap(i, j int)      { x[i], x[j] = x[j], x[i] }
+
+// Verify is the API function that verifies a verkle proofs as found in a block/execution payload.
+func Verify(vp *VerkleProof, preStateRoot []byte, postStateRoot []byte, statediff StateDiff) error {
+	// TODO: check that `OtherStems` have expected length and values.
```

Comment:
I did this TODO a while ago, [here](https://github.com/ethereum/go-verkle/blob/master/proof_ipa.go#L488-L490). I think we can remove this comment.

I also included other [similar checks](https://github.com/ethereum/go-verkle/blob/master/proof_ipa.go#L394-L406), [ordering checks](https://github.com/ethereum/go-verkle/blob/master/proof_ipa.go#L414-L417), and maybe something more I don't remember.

This angle of witness verifications (that I doubt all clients might be doing), should eventually be covered with more focused witness test-vectors. But prob those can't be execution-specs like vectors since can't be generated by "filling" tests from a correctly impl client. Anyway, big tangent here, eventually I can ask Spencer.